tests/integration/replication.tcl

```diff
@@ -405,102 +408,86 @@ test {Replica fails full sync and diskless load swapdb recovers it} {
+                    fail "Replica didn't get into loading mode"
+                }
             }
 
-            # Make sure the original keys haven't changed
-            assert_equal [$replica dbsize] 2000
-        }
-    }
-} {} {external:skip}
-
-test {Diskless load swapdb succeeds when master replid is not same (no async_loading)} {
-    start_server {tags {"repl"}} {
-        set replica [srv 0 client]
-        set replica_host [srv 0 host]
-        set replica_port [srv 0 port]
-        set replica_log [srv 0 stdout]
-        start_server {} {
-            set master [srv 0 client]
-            set master_host [srv 0 host]
-            set master_port [srv 0 port]
-
-            # Set master and replica to use diskless replication on swapdb mode
-            $master config set repl-diskless-sync yes
-            $master config set repl-diskless-sync-delay 0
-            $replica config set repl-diskless-load swapdb
+            switch $testType {
+                "Aborted" {
+                    # Make sure that next sync will not start immediately so that we can catch the replica in between syncs
+                    $master config set repl-diskless-sync-delay 5
 
-            # Put different data sets on the master and replica
-            # We need to put large keys on the master since the replica replies to info only once in 2mb
-            $replica debug populate 2000 slave 10
-            $master debug populate 500 master 100000
-            $master config set rdbcompression no
+                    # Kill the replica connection on the master
+                    set killed [$master client kill type replica]
 
-            # Set a key value on replica to check status during loading / before swapping db
-            $replica set mykey myvalue
+                    # Wait for loading to stop (fail)
+                    wait_for_condition 100 100 {
+                        [s -1 loading] eq 0
+                    } else {
+                        fail "Replica didn't disconnect"
+                    }
+                    
+                    test {Diskless load swapdb (different replid): old database is exposed after replication fails} {
+                        # Ensure we see old values from replica
+                        assert_equal [$replica get mykey] "myvalue"
 
-            # Set master with a slow rdb generation, so that we can easily intercept loading
-            # 5ms per key, with 500 keys is 2.5 seconds
-            $master config set rdb-key-save-delay 5000
+                        # Make sure amount of replica keys didn't change
+                        assert_equal [$replica dbsize] 2001
+                    }
+                }
+                "Delayed" {
+                    test {Diskless load swapdb (different replid): replica enter loading} {
+                        assert_equal [s -1 loading] 1
+                        assert_equal [s -1 async_loading] 0
+                    }
+                }
```

Comment:
sorry to bother you again, but maybe we don't need this "delayed" test? i.e. these assertions can be added into the "aborted" one before we abort it.
so this means we only have two tests "aborted" and "succeeds".

also, let's rename "fast" since the speed is just an internal concern of the test (not to take forever), but what it tests is what happens when it succeeds and fails replication..